src/core/zowe/core_for_zowe_sdk/config_file.py

```diff
@@ -0,0 +1,301 @@
+                self.location = current_dir
+                return
+
+            # check if have arrived at the root directory
+            if current_dir == os.path.dirname(current_dir):
+                break
+
+            current_dir = os.path.dirname(current_dir)
+
+        raise FileNotFoundError(f"Could not find the file {self.filename}")
+
+    def get_profilename_from_profiletype(self, profile_type: str) -> str:
+        """
+        Return exact profilename of the profile to load from the mentioned type
+
+        First tries to look into the defaults, if not found,
+        then it tries to iterate through the profiles
+        """
+        # try to get the profilename from defaults
+        try:
+            profilename = self.defaults[profile_type]
+        except KeyError:
+            warnings.warn(
+                "Given profile type has no default profilename",
+                ProfileParsingWarning,
+            )
+        else:
+            return profilename
+
+        # iterate through the profiles and check if profile is found
+        for (key, value) in self.profiles.items():
+            try:
+                temp_profile_type = value["type"]
+                if profile_type == temp_profile_type:
+                    return key
+            except KeyError:
+                warnings.warn(
+                    f"Profile {key} has no type attribute",
+                    ProfileParsingWarning,
+                )
+
+        # if no profile with matching type found, we raise an exception
+        raise ProfileNotFound(
+            profile_name=profile_type,
+            error_msg=f"No profile with matching profile_type {profile_type} found",
+        )
+
+    def load_profile_properties(self, profile_name: str) -> dict:
+        """
+        Load exact profile properties (without prepopulated fields from base profile)
+        from the profile dict and populate fields from the secure credentials storage
+        """
+        try:
+            props = self.profiles[profile_name]["properties"]
+        except Exception as exc:
+            raise ProfileNotFound("Profile {profile_name} not found", error_msg=exc)
+
+        secure_fields: list = self.profiles[profile_name].get("secure", [])
+
+        # load secure props only if there are secure fields
+        if secure_fields:
+            self.load_secure_props()
+
+            # load properties with key as profile.{profile_name}.properties.{*}
+            for (key, value) in self.secure_props.items():
+                if re.match(
+                    "profiles\\." + profile_name + "\\.properties\\.[a-z]+", key
+                ):
+                    property_name = key.split(".")[3]
+                    if property_name in secure_fields:
+                        props[property_name] = value
+                        secure_fields.remove(property_name)
+
+            if len(secure_fields) > 0:
+                raise SecureValuesNotFound(secure_fields)
```

Comment:
Instead of raising this error at the config file level, can we delay throwing a `SecureValuesNotFound` error until we have loaded all the config files in `ProfileManager.load`?

It should be allowable to store "user" and "password" in a zowe.config.user.json file instead of in the secure vault 🙂

Reply:
After the above comment is addressed, we can verify that the `SecureValuesNotFound` error is issued when secure credentials are missing.

Reply:
@shalearkane I updated some of the logic for merging configs and profiles to simplify it a bit 🙂 I believe the comment above still needs to be addressed. Please let me know if you have any questions.